### PR TITLE
Add RabbitMq variables mapping 

### PR DESCRIPTION
**Pull request by maximgubar**

### CODE REVIEW

File: platformsh-flex-env.php [hunk 1]
```diff
@@ -47,6 +47,7 @@ function mapPlatformShEnvironment() : void
     // Map services as feasible.
     mapPlatformShDatabase('database', $config);
     mapPlatformShMongoDatabase('mongodatabase', $config);
+    mapPlatformShRabbitMq('rabbitmq', $config);
```
Comment (Crell) [inline platformsh-flex-env.php]: We *just* (like, on Monday) decided on an internal convention for relationship names. :-)  Can you rename this to `rabbitmqqueue`?

**Comment by Crell** [comment]
This will need unit tests, too.  Thanks!

File: platformsh-flex-env.php (hunk 1 above)
Comment (maximgubar) [reply platformsh-flex-env.php]: sure, done.

**Comment by maximgubar** [comment]
seems like I've messed a bit with commits, shall I create new MR ?

**Comment by Crell** [comment]
You can rebase the branch if you want to clean up the history, or I can do a squash merge when it gets merged.  Either works.

However, we're trying to avoid using PHPUnit's mock library.  Have a look at the other tests in the library right now; they're using testdata files to load up actual classes.  That also lets us have a clearer picture of how the library behaves in "real" situations, like on someone's local laptop, where the env vars are in oddball states.

**Comment by Plopix** [comment]
What if we have more than one queue? Then we should have more than 1 DSN.
So I guess the last part "messages" should be injected differently?

**Comment by Crell** [comment]
@Plopix How so?  I don't think we can realistically auto-map an undefined number of relationships.  The automapping here is for the common cases.  If you want to do something more complex, you will almost certainly need to do your own mapping logic.

How can we ensure the defaults don't make the more complex cases *harder*, even if they don't help with them directly?

**Comment by Plopix** [comment]
Hey @crell,

I get your point I was just pointing out a difference with the other auto-mapped variables.
If I am not mistaken the bridge overrides the variable even if it exists for all of them.
And that's fine for MySQL, MongoDB, SwiftMailer.

But if we don't want to use the `messages` queue MESSENGER_TRANSPORT_DSN will override anyway.

And I am not sure that this is great, I don't know I wanted to share that.

I guess in this situation we could just use another `relationship name` and we would not be impacted by the bridge for this relationship. (which makes my comments useless)

But, as a feedback, that would be awesome if the bridge could somehow look somewhere and create all the vars needed for all the project queues.

Thanks for reading me ;)

**Comment by Crell** [comment]
I agree, it would be nifty if we could auto-map an arbitrary number of queues.  I have no idea how we'd do that, though. :smile:  Especially if we want to avoid a situation that breaks if you go even slightly off of the path we specify.

Perhaps providing the DSN through a formatter that users could reuse themselves if they wanted to would be better?

**Comment by Plopix** [comment]
I don't know for the DSN, not sure I got it perfectly. That would be a way to inject some code from outside when you create the ENV var?

For eZ Launchpad and to handle multi-database imports/exports we had this "kind of" issue.
To know how to loop to export/import those databases we added an ENV variable that lists the information we needed.

So that could be a way. The user of the bridge could provide an ENV var, let's say. 
`QUEUE_NAMES=something index fileprocessing` and the bridge would check that var, and loops on it to create 3 transports.

That requires to create an new ENV var to create many ENV vars. It is not perfect but that would work and could be generic for other usage.

File: platformsh-flex-env.php [hunk 2]
```diff
@@ -228,3 +229,21 @@ function mapPlatformShMongoDatabase(string $relationshipName, Config $config): v
...
+function mapPlatformShRabbitMq(string $relationshipName, Config $config): void
+{
+    if (!$config->hasRelationship($relationshipName)) {
+        return;
+    }
+
+    $credentials = $config->credentials($relationshipName);
+
+    setEnvVar('MESSENGER_TRANSPORT_DSN', sprintf(
+        '%s://%s:%s@%s:%d/%%2f/messages',
```
Comment (Plopix) [inline platformsh-flex-env.php:242]: `%2f` should be dynamic as we can have a vhost on Dedicated

```suggestion
        '%s://%s:%s@%s:%d/%s/messages',
```

File: platformsh-flex-env.php [hunk 3]
```diff
@@ -228,3 +229,21 @@ function mapPlatformShMongoDatabase(string $relationshipName, Config $config): v
...
+        $credentials['scheme'],
+        $credentials['username'],
+        $credentials['password'],
+        $credentials['host'],
+        $credentials['port']
```
Comment (Plopix) [inline platformsh-flex-env.php:247]:
```suggestion
        $credentials['port'],
        $credentials['vhost'] ?? '%2f'
```

Comment (Crell) [reply platformsh-flex-env.php:247]: There is no vhost parameter in the credentials array, is there?

File: platformsh-flex-env.php [hunk 4]
```diff
@@ -228,3 +229,21 @@ function mapPlatformShMongoDatabase(string $relationshipName, Config $config): v
...
+
+function mapPlatformShRabbitMq(string $relationshipName, Config $config): void
+{
+    if (!$config->hasRelationship($relationshipName)) {
+        return;
+    }
+
+    $credentials = $config->credentials($relationshipName);
+
+    setEnvVar('MESSENGER_TRANSPORT_DSN', sprintf(
```
Comment (Crell) [inline platformsh-flex-env.php:241]: This should be turned into a reusable formatter.

File: tests/FlexBridgeRabbitMqTest.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+class FlexBridgeRabbitMqTest extends TestCase
+{
+    /**
+     * @var MockObject|Config
+     */
+    private $configMock;
+
+    public function setUp(): void
+    {
+        $this->configMock = $this->getMockBuilder(Config::class)
```
Comment (Crell) [inline tests/FlexBridgeRabbitMqTest.php:21]: Please use the pattern set in the other tests.  We are not using the PHPUnit mock library.

File: platformsh-flex-env.php (hunk 3 above)
Comment (Plopix) [reply platformsh-flex-env.php:247]: not on PS, but on Dedicated you have one ;-)

**Pull request closed without merging**